app/src/services/auth/authService.js

```diff
@@ -0,0 +1,132 @@
+"use strict";
+
+const UserRepository = require("../../repositories/user/userRepository");
+const JwtService = require("../../utils/jwtService");
+const bcrypt = require("bcrypt");
+
+class AuthService {
+  constructor(req) {
+    this.req = req;
+    this.body = req.body;
+    this.userRepository = new UserRepository();
+    this.jwtService = new JwtService();
+  }
+
+  async signUp() {
+    const { user_id, password, nickname } = this.body;
+
+    const isUserIdExist = await this.userRepository.findByUserId(user_id);
+    const isNicknameExist = await this.userRepository.findByNickname(nickname);
+
+    if (isUserIdExist && isNicknameExist) {
+      return { status: 409, success: false, data: { field: "both" } };
+    } else if (isUserIdExist) {
+      return { status: 409, success: false, data: { field: "user_id" } };
+    } else if (isNicknameExist) {
+      return { status: 409, success: false, data: { field: "nickname" } };
+    }
+
+    const hashedPassword = await bcrypt.hash(password, 10);
+
+    const createdUser = await this.userRepository.createUser({
+      user_id,
+      password: hashedPassword,
+      nickname,
+    });
+
+    return {
+      status: 201,
+      success: true,
+      data: {
+        field: null,
+        message: "회원가입 성공",
+        user: {
+          user_id: createdUser.user_id,
+          nickname: createdUser.nickname,
+        },
+      },
+    };
+  }
+
+  async login() {
+    const { user_id, password } = this.body;
+
+    const user = await this.userRepository.findByUserId(user_id);
+    if (!user) {
+      return {
+        status: 404,
+        success: false,
+        data: { message: "존재하지 않는 아이디입니다." },
```

Comment:
동작에는 영향이 없지만 추후 디버깅할 때 유용할 수 있어요

```suggestion
        data: { message: "존재하지 않는 유저입니다." }, // 또는 아이디에 해당하는 유저가 존재하지 않습니다
```

Reply:
피드백 감사합니다!
말씀해주신 대로 메시지를 좀 더 구체적으로 작성하는 것이 추후 디버깅이나 사용자 이해에 도움이 될 것 같습니다👍